src/data/list/forall2.lean

```diff
@@ -167,6 +167,18 @@ lemma rel_append : (forall₂ r ⇒ forall₂ r ⇒ forall₂ r) append append
 | [] [] h l₁ l₂ hl := hl
 | (a::as) (b::bs) (forall₂.cons h₁ h₂) l₁ l₂ hl := forall₂.cons h₁ (rel_append h₂ hl)
 
+lemma rel_reverse : (forall₂ r ⇒ forall₂ r) reverse reverse
+| [] [] forall₂.nil := forall₂.nil
+| (a::as) (b::bs) (forall₂.cons h₁ h₂) :=
+  by simp only [reverse_cons];
+  from rel_append (rel_reverse h₂) (forall₂.cons h₁ forall₂.nil)
+
+@[simp]
+lemma forall₂_reverse_iff {l₁ l₂} : forall₂ r (reverse l₁) (reverse l₂) ↔ forall₂ r l₁ l₂ :=
+iff.intro
+  (assume h, by rw [← reverse_reverse l₁, ← reverse_reverse l₂]; from rel_reverse h)
```

Comment:
mathlib tends to prefer avoiding semicolons unless there are actually multiple goals:
```suggestion
  (assume h, by { rw [← reverse_reverse l₁, ← reverse_reverse l₂], exact rel_reverse h })
```

Reply:
👍